### DISCUSSION TITLE
libva and libva-drm

### DISCUSSION THREAD

**Discussion by rubikstriangle** [AcademySoftwareFoundation/OpenRV#92] [Build Issues]
After building OpenRV on centos7 and testing that build in Rocky Linux and Ubuntu I found that I was missing libva and libva-drm.  Copying these libs from centos7 into my rv lib folder seems to fix the issue.  As a more permanent solution I'm wondering if it would make sense to include these libs in OpenRv by default?

**Comment by rubikstriangle** [comment]
Actually, I tested this out about a week ago and I thought that I had found that libva and libva-drm weren't available on Rocky.  I just double checked and found the they are available, they are just one version higher, so libva.so.2 instead of libva.so.1.  In this case is it better to create a symbolic link from /usr/lib64/libva.so.2 to /some/dir/rv/lib/libva.so.1 or is it better to use the libs from centos 7.  Both seem to work but I haven't thoroughly tested them.

**Reply by rubikstriangle** [reply]
...or would it be better to just compile OpenRV on Rocky Linux 8.7

**Reply by geffrak** [reply]
Compiling on the same Linux distribution as the targeted one is always easier. Otherwise, you need to know what to include and avoid, and it's a trial-and-error process.

**Comment by geffrak** [comment]
What version of Rocky Linux are you using?

**Reply by rubikstriangle** [reply]
Rocky Linux 8.7

**Comment by rith88** [comment]


**Reply by rubikstriangle** [reply]
You need to have openCL libraries installed.  If you have the Nvidia graphics drivers installed it should take care of this.  I recently did a build on a virtual machine without a graphics card and found that force installing the Nvidia drivers did the trick.
I also needed to install ninja-devel and meson using yum.  Seems like Ninja also comes with QT but it didn't seem to work out of the box for me but the ninja-devel seemed to make the build happy.  

I also found that the libva-drm and libra libs are missing on other flavors of linux so I packed it up with my install.  This makes Rocky Linux happy.
```shell
cp /lib64/libva-drm.so.1.4000.0 /openRV/lib/libva-drm.so.1
cp /lib64/libva.so.1.4000.0 /openRV/lib/libva.so.1
```

**Reply by rith88** [reply]
Thanks a lot, it worked

**Comment by geffrak** [comment]
This is being adressed in #315
